Automattic/wp-calypso
# E2E: Fix failing `specs/editor/editor__navbar.ts` test due to recent removal of editor nav sidebar

Resolves https://github.com/Automattic/dotcom-forge/issues/7908.
Related to https://github.com/Automattic/dotcom-forge/issues/7905, D153130-code and https://github.com/Automattic/jetpack/pull/37893.

## Proposed Changes

The aim of this PR is to update the `specs/editor/editor__navbar.ts` test to reflect the current user experience where the editor's nav sidebar is no longer present:

![Markup on 2024-06-24 at 12:54:16](https://github.com/Automattic/wp-calypso/assets/25105483/3d8658e8-7999-479b-b812-dc1ff013ea5f)

Instead of this sidebar, the user is directed to the dashboard as soon as the `W` logo in the editor's corner is clicked:

![Markup on 2024-06-24 at 12:54:59](https://github.com/Automattic/wp-calypso/assets/25105483/605214c9-bb51-434f-bf7c-e07b5fbc25cf)

Details:

* Introduce new `editorToolbarComponent.closeEditor()` action
* remove the following logic (as it is no longer needed):
  * `editorNavSidebarComponent` from `editor-page.ts`
  * `whole editor-nav-sidebar-component.ts`
  * `openNavSidebar()` and `closeNavSidebar()` from `editor-toolbar-component.ts`
  * `editor-nav-sidebar-component` reference from E2E tests components `index.ts` 

Please note that the E2E tests related to Simple sites will fail until D153130-code is merged and deployed to production.

## Why are these changes being made?


The changes are being made to fix failing `specs/editor/editor__navbar.ts` test.

## Testing Instructions



1. Check out the branch.
2. Navigate to `tests/e2e` and run `yarn build --watch`.
3. In a separate terminal window run `VIEWPORT_NAME=desktop TEST_ON_ATOMIC=true GUTENBERG_EDGE=false yarn jest specs/editor/editor__navbar.ts`.
4. The test should pass.
5. Replace the `VIEWPORT_NAME=desktop` with `VIEWPORT_NAME=mobile`, `TEST_ON_ATOMIC=false` with `TEST_ON_ATOMIC=true` and `GUTENBERG_EDGE=false` with `GUTENBERG_EDGE=true` (try different combinations).
  - when testing with Simple test sites, the D153130-code patch needs to be applied and sites mentioned on the bottom of this patch description need to be sandboxed.
7. All the tests should pass.

You may also find the following instructions to set up E2E testing environment helpful: https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/README.md.

## Pre-merge Checklist
